Geometry_Engine/Query/Length.cs

```diff
@@ -51,41 +59,114 @@ public static double SquareLength(this Vector vector)
         /**** Public Methods - Curves                   ****/
         /***************************************************/
 
+        [Description("Calculates the length of an Arc.")]
+        [Input("curve", "The Arc to calculate the length of.")]
+        [Output("length", "The length of the Arc.", typeof(Length))]
         public static double Length(this Arc curve)
         {
             return curve.Angle() * curve.Radius;
         }
 
         /***************************************************/
 
+        [Description("Calculates the length of an Circle.")]
+        [Input("curve", "The Circle to calculate the length of.")]
+        [Output("length", "The length of the Circle.", typeof(Length))]
         public static double Length(this Circle curve)
         {
             return 2 * Math.PI * curve.Radius;
         }
 
         /***************************************************/
 
+        [Description("Calculates the length of an Ellipse. Evaluated as an infinite series, utilising 10 times the ratio of the long and short radius number of terms.\n" +
+                     "Gives very close to exact results for ellipses with an ratio of up to 1:20 000 between the long and short radius.")]
+        [Input("curve", "The Ellipse to calculate the length of.")]
+        [Output("length", "The length of the Ellipse.", typeof(Length))]
+        public static double Length(this Ellipse curve)
+        {
+            //Get out a as the long radius and b as the short radius
+            double a = Math.Max(curve.Radius1, curve.Radius2);
+            double b = Math.Min(curve.Radius1, curve.Radius2);
+
+            //Special case circle
+            if (a == b)
+                return 2 * Math.PI * a;
+
+            //Special case line
+            if (b == 0)
```

Comment:
Wouldn't it be worth applying tolerance here?

Reply:
Same as above. This is mainly here for performance. Guess I could check if b/a < 1e-9 or something here though